method/Giotto/Giotto.r

```diff
@@ -0,0 +1,289 @@
+
+
+
+## raw expression counts expected
+createGiotto_fn = function(spe, annotation = FALSE, selected_clustering = NULL, instructions = NULL){
+  raw_expr <- SummarizedExperiment::assay(spe, "counts")
+  #colnames(raw_expr) <- colData(sce)[,"Barcode"]
+  norm_expression <- SummarizedExperiment::assay(spe, "logcounts")
+  
+  cell_metadata <- SingleCellExperiment::colData(spe)
+  cell_metadata$cell_ID <- rownames(SingleCellExperiment::colData(spe))
+  colnames(cell_metadata)[c(1,2)] <- c("sdimx", "sdimy")
+  cell_metadata <- as.data.frame(cell_metadata[,c(4,1,2,3)])
+  feat_metadata <- tibble::as_tibble(SingleCellExperiment::rowData(spe),rownames = "feat_ID")
+  #colnames(feat_metadata)[[1]] <- c("feat_ID")
+  #rownames(raw_epxr) <- gene_metadata$gene_ID
+  if (annotation) {
+    rownames(raw_expr) <- c(SingleCellExperiment::rowData(spe)[, "SYMBOL"])
+    #rownames(norm_expression) <- c(SingleCellExperiment::rowData(sce)[,"SYMBOL"])
+  }
+  gobj = Giotto::createGiottoObject(
+      expression = list("raw" = raw_expr,
+                        "normalized" = norm_expression),
+      cell_metadata = cell_metadata,
+      spatial_locs = as.data.frame(SpatialExperiment::spatialCoords(spe)),
+      feat_metadata = feat_metadata,
+      instructions = instructions#,
+      # Add dimred (doesn't quite work)
+      #dimension_reduction = GiottoClass::createDimObj(
+      #    coordinates = SingleCellExperiment::reducedDim(spe, "reducedDim"),
+      #    name = "PCA",
+      #    method = "pca")
+  )
+  return(gobj)
+}
+# Convert to Giotto object
+gobj <- createGiotto_fn(spe, instructions = instrs)
+
+# Normalize
+# gobj <- Giotto::normalizeGiotto(gobj)
+# Alternatively, use the Giotto normalization
+
+# PCA
+gobj <- runPCA(gobject = gobj, center = TRUE, scale_unit = TRUE, name = "PCA", method = "exact", feats_to_use = NULL)
+# Compatibility issues with Matrix and method = "irlba" - the fix is to downgrade Matrix to 1.6.1, but GiottoClass requires 1.6.2
+# Using provided PCA throws:
+# (Error in cov(y[lclust[[i]], ]) :
+#   supply both 'x' and 'y' or a matrix-like 'x')
+
+
+
+### Run clustering
+
+#if(!file.exists(hmrf_folder)) dir.create(hmrf_folder, recursive = T)
+message("Running ", method, " clustering")
+
+# create spatial network (tricky to use input network)
+gobj <- Giotto::createSpatialNetwork(
+    gobject = gobj,
+    k = k
+    #minimum_k = 10
+)
+
+# identify genes with a spatial coherent expression profile (Not necessary - default uses all 'selected' features)
+#km_spatialgenes <- Giotto::binSpect(gobj, bin_method = 'rank')
+
+#my_spatial_genes <- km_spatialgenes[1:100]$feats
+HMRF_spatial_genes <- Giotto::doHMRF(
+    gobject = gobj,
+    spat_unit = "cell",
+    feat_type = "rna",
+    betas = c(0, 2, config$beta),
+    expression_values = "normalized",
+    spatial_genes = NULL, #my_spatial_genes,
+    dim_reduction_to_use = "pca",
+    dim_reduction_name = "PCA",
+    dimensions_to_use = 1:20,
```

Comment:
If we cant use provided dimensionality reduction, maybe the number of dims should be a config parameter